src/radical/pilot/session.py

```diff
@@ -254,13 +258,21 @@ def close(self, cleanup=False, terminate=True, download=False):
         self._log.debug("session %s closing", self._uid)
         self._prof.prof("session_close", uid=self._uid)
 
-        # set defaults
-        if cleanup   is None: cleanup   = True
-        if terminate is None: terminate = True
-
-        if  cleanup:
-            # cleanup implies terminate
-            terminate = True
+        # Merge kwargs with current defaults stored in self._close_options
+        self._close_options.update(
+            {
+                'cleanup': cleanup,
+                'download': download,
+                'terminate': terminate
+            }
+        )
```

Comment:
I think, it is better to use keyword arguments here, thus if certain argument is set then only the corresponding close-option will be updated and others will keep their default values
```python
    def close(self, **kwargs):
        ...
        self._close_options.update(kwargs)
        ...
# s.close(download=True) -> cleanup: False, download: True, terminate: True
```
(*) `close(hello=True)` will raise an error, that `hello` is not in schema

Reply:
Done.